File: backend/schemas/recommendation.py
from datetime import datetime
from pydantic import BaseModel

class RecommendationBase(BaseModel):
    crime_event_id: int | None = None
    priority: str = "medium"
    recommendation_text: str
    reason: str | None = None
    status: str = "pending"
    confidence: float | None = 0.80
    supporting_analytics: str | None = None

class RecommendationCreate(RecommendationBase):
    pass

class RecommendationResponse(RecommendationBase):
    id: int
    created_at: datetime

    model_config = {
        "from_attributes": True
    }


class AllocationPayload(BaseModel):
    district: str
    sanctioned_asi: int
    sanctioned_chc: int
    sanctioned_cpc: int


class BeatAllocation(BaseModel):
    beat_name: str
    asi_allocated: int
    chc_allocated: int
    cpc_allocated: int
    normalized_severity: float


class AllocationResponse(BaseModel):
    status: str
    district: str
    solved_allocation: list[BeatAllocation]


class ResourceAllocationResponse(BaseModel):
    id: int
    district: str
    allocated_asi: int
    allocated_chc: int
    allocated_cpc: int
    solved_allocation: list[BeatAllocation]
    created_at: datetime

    model_config = {
        "from_attributes": True
    }


class RecommendationStatusUpdate(BaseModel):
    status: str


class RecommendationHistoryResponse(BaseModel):
    id: int
    dataset_ids: str
    model_version: str
    alert_count: int
    generated_recommendations_count: int
    status: str
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/types/recommendation.ts
export interface Recommendation {
  id: number;
  crime_event_id: number | null;
  priority: "high" | "medium" | "low";
  recommendation_text: string;
  reason: string | null;
  status: "pending" | "resolved" | "dismissed";
  created_at: string;
}

export interface BeatAllocation {
  beat_name: string;
  asi_allocated: number;
  chc_allocated: number;
  cpc_allocated: number;
  normalized_severity: number;
}

export interface AllocationPayload {
  district: string;
  sanctioned_asi: number;
  sanctioned_chc: number;
  sanctioned_cpc: number;
}

export interface AllocationResponse {
  status: string;
  district: string;
  solved_allocation: BeatAllocation[];
}

export interface ResourceAllocation {
  id: number;
  district: string;
  allocated_asi: number;
  allocated_chc: number;
  allocated_cpc: number;
  solved_allocation: BeatAllocation[];
  created_at: string;
}

export interface DecisionSupportMetrics {
  totalCount: number;
  pendingCount: number;
  resolvedCount: number;
  dismissedCount: number;
}
